devito/types/utils.py

```diff
@@ -24,6 +24,12 @@ class Stagger(Tag):
 
 class DimensionTuple(EnrichedTuple):
 
+    __rkwargs__ = EnrichedTuple.__rkwargs__ + ('left', 'right')
+
+    def __new__(cls, *args, getters=None, left=None, right=None, **kwargs):
```

Comment:
if we really have to add this `__new__`, which basically doesn't add anything to the super()'s `__new__`, then it looks odd

Reply:
Doesn't need to be added to `__new__` so those parameters are explicit, but it's not strictly necessary since that's covered by `self.__dict__.assign(...` in the parent. I can remove the parameters here if preferred